src/mds/Server.cc

```diff
@@ -6210,6 +6222,24 @@ void Server::handle_client_mkdir(MDRequestRef& mdr)
   mds->balancer->maybe_fragment(dir, false);
 }
 
+void Server::set_dentry_alternate_name(CDentry *dn, const cref_t<MClientRequest>& req)
+{
+  if (!req->get_data().length())
+    return;
+
+  auto p = req->get_data().cbegin();
+
+  // Try to find whether there has a fake xattr named
+  // as "alternate_name"
+  auto _xattrs = CInode::allocate_xattr_map();
+  decode_noshare(*_xattrs, p);
+
+  auto it = _xattrs->find("alternate_name");
+  if (it != _xattrs->end()) {
+    dn->set_alternate_name(std::move(it->second));
+    dout(10) << "dn: " << dn << " has a alternate name!" << dendl;
```

Comment:
Let's move this debug message to the `CDentry::set_alternate_name` method. and adjust the wording:

```suggestion
    dout(10) << "setting alternate_name on " << dn << dendl;
```